Makefile

```diff
@@ -212,7 +215,51 @@ check-license-headers:
 			exit 1 ; \
 		fi ; \
 	done ; \
-	echo "" ; echo "✓ All code files have the required license header." ; echo ""
+	echo "" ; echo "\033[1;32m✓\033[0m All code files have the required license header." ; echo ""
+
+.PHONY: check-fmtprints
+check-fmtprints: SHELL:=/bin/sh
+check-fmtprints: # fails if there are any fmt.Print* calls outside of the 3 approved files
+	@cd pkg && \
+	fmtprints=$$(git grep -n fmt.Print | grep -v 'appinfo/usage/usage.go' | grep -v 'config/validate/validate.go' | grep -v '^daemon/'); \
+	count=0; \
+	if [ -n "$$fmtprints" ]; then \
+		count="$$(echo "$$fmtprints" | wc -l | tr -d '[:space:]')" ; \
+	fi; \
+	if [ "$$count" -ne 0 ]; then \
+		echo "" ; \
+		echo "\033[1;31m⨉\033[0m ($$count) unexpected fmt.Print*(s) must be removed from the codebase:"; \
+		echo "" ; \
+		echo "$$fmtprints" ; \
+		echo "" ; \
+		echo "" ; \
+		exit 1; \
+	fi ; \
+	echo "" ; echo "\033[1;32m✓\033[0m No unexpected fmt.Print* calls." ; echo ""
+
+.PHONY: check-todos
+check-todos: SHELL:=/bin/sh
+check-todos: # fails if there are any fmt.Print* calls outside of the 3 approved files
+	@cd pkg && \
+	todos=$$(git grep -in todo | grep -v 'context\.TODO'); \
+	count=0; \
+	if [ -n "$$todos" ]; then \
+		count="$$(echo "$$todos" | wc -l | tr -d '[:space:]')" ; \
```

Comment:
fyi, golangci-lint can do this: https://golangci-lint.run/usage/linters/#godox

Reply:
Currently we have linting of `*_test.go` disabled (enabling it would create hundreds of new issues), and I want to pick up the 36% of TODOs in the test files. I would propose switching this to the linter once we have burned down the current list to 0 and have the linter enabled for tests too.